### PR TITLE
Release: copy sizzle to external folder in dist, remove files during dist copy

### PR DESCRIPTION
**Pull request by timmywil**
### Summary ###

Fixes a couple release issues. Tested with jquery-release.

Fixes gh-3094
Fixes gh-2945

### Checklist ###

* [x] All authors have signed the CLA at https://contribute.jquery.com/CLA/
* ~~[ ] New tests have been added to show the fix or feature works~~
* [x] Grunt build and unit tests pass locally with these changes
* ~~[ ] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com~~

### CODE REVIEW

File: build/release/dist.js [hunk 1]
```diff
@@ -56,12 +55,31 @@ module.exports = function( Release, files, complete ) {
...
 
 		// Copy dist files
-		var distFolder = Release.dir.dist + "/dist";
+		var distFolder = Release.dir.dist + "/dist",
+			externalFolder = Release.dir.dist + "/external",
+			rmIgnore = files.concat( [ "README.md", "node_modules" ] ),
+			i = rmIgnore.length;
+
+		while ( --i ) {
+			rmIgnore[ i ] = Release.dir.dist + "/" + rmIgnore[ i ];
```
Comment (mgol) [inline build/release/dist.js]: Why not
```js
rmIgnore = rmIgnore.map( function( dirname ) {
    return Release.dir.dist + "/" + dirname;
} );
```
? The `i` variable would be unnecessary.

**Comment by mgol** [comment]
From what I see part of it has already partially landed on `1.12-stable` in 59003ae7132677023247ed1868e4e426e4c2fe81 (but not on `2.2-stable` or `master`)?

File: build/release/dist.js (hunk 1 above)
Comment (timmywil) [inline build/release/dist.js]: Do all the nodes we support have map?

**Comment by timmywil** [comment]
Hmm. That's odd. I don't remember pushing that.

File: build/release/dist.js (hunk 1 above)
Comment (timmywil) [inline build/release/dist.js]: Ok, 0.10 has it. I thought it didn't for some reason. I'll make the switch.

Comment (markelog) [inline build/release/dist.js]: Keep in mind that this code only be used by the maintainers only :)

**Comment by timmywil** [comment]
@mgol My mistake. I must have had the commit on that branch locally for a minute. It's been reverted until this PR is merged.

File: build/release/dist.js (hunk 1 above)
Comment (mgol) [inline build/release/dist.js]: Yeah, Node.js has been ES5-compatible for a long time now. :)

**Pull request closed without merging**